src/dashboard/src/components/filesystem_ajax/views.py

```diff
@@ -1103,7 +1104,14 @@ def download_by_uuid(request, uuid, preview_file=False):
             % {"uuid": uuid},
         }
         return helpers.json_response(response, status_code=404)
-    relative_path = f.currentlocation.replace("%transferDirectory%", "")
+    # check if the transfer is in the backlog and if it's a bag
+    transfer = models.Transfer.objects.get(uuid=f.transfer.uuid)
+    transfer_name = os.path.basename(transfer.currentlocation.rstrip("/"))
+    transfer_dir = os.path.join(ORIGINAL_DIR, transfer_name)
+    if os.path.isdir(transfer_dir) and is_bag(transfer_dir):
```

Comment:
This feels a bit hacky and I take blame for that...

I wonder if `Transfer` should learn if it's in backlog and its format. Something to consider in the future.

Reply:
It does feel hacky but couldn't find an alternative.